retrofit/src/main/kotlin/com/smartnsoft/ws/exception/CallException.kt

```diff
@@ -0,0 +1,67 @@
+package com.smartnsoft.ws.exception
+
+import java.io.InterruptedIOException
+import java.net.SocketException
+import java.net.SocketTimeoutException
+import java.net.UnknownHostException
+import javax.net.ssl.SSLException
+
+/**
+ * @author Anthony Msihid
+ * @since 2019.07.18
+ */
+
+/**
+ * The exception that will be thrown if any problem occurs during a web service call.
+ */
+@Suppress("unused")
+open class CallException
+@JvmOverloads
+constructor(message: String? = null, throwable: Throwable? = null, val statusCode: Int = 0)
+  : Exception(message, throwable)
+{
+
+  constructor(throwable: Throwable? = null) : this(null, throwable, 0)
```

Comment:
Since you have @JvmOverloads on the main constructor with default values, is it really useful to have these 3 additional constructors ?